crates/eth_compatibility/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![no_std]
+use ink_env::{
+    DefaultEnvironment,
+    Environment,
+};
+
+/// The ECDSA compressed public key.
+#[derive(Debug, Copy, Clone)]
+pub struct ECDSAPublicKey(pub [u8; 33]);
+
+impl Default for ECDSAPublicKey {
+    fn default() -> Self {
+        // Default is not implemented for [u8; 33], so we can't derive it for ECDSAPublicKey
+        // But clippy thinks that it is possible. So it is workaround for clippy.
+        let empty = [0; 33];
+        Self { 0: empty }
+    }
+}
+
+impl core::ops::Deref for ECDSAPublicKey {
+    type Target = [u8; 33];
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl core::ops::DerefMut for ECDSAPublicKey {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// The address of an Ethereum account.
+#[derive(Debug, Default, Copy, Clone)]
+pub struct EthereumAddress(pub [u8; 20]);
+
+impl core::ops::Deref for EthereumAddress {
+    type Target = [u8; 20];
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl core::ops::DerefMut for EthereumAddress {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
```

Comment:
`Deref` and `DerefMut` should be used with great care and usually should only be applied on reference-like or pointer-like types.
Instead I'd propose to implement `AsRef<[u8; 33]>` and `AsMut<[u8; 33]>`.
This leads the same purpose but is way less intrusive.